### PR TITLE
[AIRFLOW-XXXX]  Use on_load in plugin example

### PR DESCRIPTION
**Pull request by drewsonne**
Rather than instantiating objects on file load, the plugin example should encourage instantiation at the correct time of the lifecycle, by encapsulating those calls in the on_load method.

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Modified the Plugin documentation to encourage use of `on_load` to perform actions as part of the plugin lifecycle, rather than when the module is loaded.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Documentation

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by Fokko** [comment]
Calling in @ashb, our plugin expert

**Review by ashb — COMMENTED** [review]
Hmmmm.

Given that all the views/menus/etc are static in this example I think I'd prefer we show people the "default" example of just assigning class properties like before

**Comment by drewsonne** [comment]
My motivation here is to try to start to build a specific plugin lifecycle.

The existing example instantiates the component objects (TestAppBuilderBaseView , Blueprints, Views etc) when the module is loaded. This means the plugin author can't rely objects/classes being available when their plugin is loaded (eg, flask-app/config) as changes are made to Airflow between releases. 

I don't have a wider proposal for what the full plugin lifecycle would be, but I think encouraging all the imperative actions to be encapsulated in the on_load method, reduces the risk of plugins breaking because a component the author may be interacting with does not yet exist.

Sort of providing developers with a predictable safe way to bootstrap the components in their plugins.

I appreciate the example does not fall pray to these issues, but I'm suggesting our example should provide "best practice" and I'm suggesting what our best practice should be. :-)

If you'd still prefer not, I'll close the PR. :-)

**Comment by drewsonne** [comment]
@ashb technically they're still class properties ;-)

**Comment by ashb** [comment]
That's a good reason - so can we have something to that effect in the doc?

**Comment by OmerJog** [comment]
@drewsonne can you resolve conflict?
Also, shouldn't this have Jira ticket? It has code changes.

File: docs/plugins.rst [hunk 1]
```diff
@@ -199,16 +190,6 @@ definitions in Airflow.
...
-                            "view": v_appbuilder_view}
-
-    # Creating a flask appbuilder Menu Item
-    appbuilder_mitem = {"name": "Google",
-                        "category": "Search",
-                        "category_icon": "fa-th",
-                        "href": "https://www.google.com"}
 
     # Defining the plugin class
     class AirflowTestPlugin(AirflowPlugin):
```
Comment (mik-laj) [inline docs/plugins.rst:195]: This file is already in the repository. Can you add a reference to this file using the literalinclude directive? Or even better - you can use the exampleinclude directive to add a button to the full source code
https://github.com/apache/airflow/blob/d333ffa5cc4f79ccccf36117d4b33b546da9091e/tests/plugins/test_plugin.py

**Comment by drewsonne** [comment]
@OmerJog there are no changes in "code" for the functionality of the airflow product, nor its tests. The changes are in a markdown format for the readme

**Pull request closed without merging**